app/views/venues/show.html.erb

```diff
@@ -1,15 +1,30 @@
 <h1><%= @venue.venue_name %></h1>
 <p><%= @venue.description %></p>
+
+<% if @venue.deletable_by(current_user) %>
+  <%= button_to "Delete Venue", venue_path, class: "button", method: "delete", label: "Delete Venue"%>
+<% end %>
+<% if @venue.deletable_by(current_user) %>
+  <%= button_to "Edit Venue", edit_venue_path, class: "button", method: "get", label: "Edit Venue"%>
+<% end %>
+
 <ul>
   <% @reviews.each do |review| %>
-  <li> <%= review.review_body %></li>
-  <%= review.rating %>
+    <li> <%= review.review_body %></li>
+    <%= review.rating %>
+    <% if review.deletable_by(current_user) %>
+      <%= button_to "Delete Review", "#{@venue.id}/reviews/#{review.id}", class: "button", method: "delete", label: "Delete Review" %>
```

Comment:
I would use the path helper, here.

It's going to be something like: `venue_reviews_path(@venue, review)`

Use the `rake routes` command to find out.